compass/landice/tests/greenland/mesh.py

```diff
@@ -203,18 +139,22 @@ def run(self):
                                 'westCentralGreenland'],
                           all_tags=False)
 
+        # Do some final validation of the mesh
+        ds = xr.open_dataset(self.mesh_filename)
         # Ensure basalHeatFlux is positive
-        data.variables['basalHeatFlux'][:] = np.abs(
-            data.variables['basalHeatFlux'][:])
+        ds["basalHeatFlux"] = np.abs(ds.basalHeatFlux)
         # Ensure reasonable dHdt values
-        dHdt = data.variables["observedThicknessTendency"][:]
-        dHdtErr = data.variables["observedThicknessTendencyUncertainty"][:]
+        dHdt = ds["observedThicknessTendency"]
         # Arbitrary 5% uncertainty; improve this later
         dHdtErr = np.abs(dHdt) * 0.05
-        # large uncertainty where data is missing
-        dHdtErr[np.abs(dHdt) > 1.0] = 1.0
-        dHdt[np.abs(dHdt) > 1.0] = 0.0  # Remove ridiculous values
-        data.variables["observedThicknessTendency"][:] = dHdt
-        data.variables["observedThicknessTendencyUncertainty"][:] = dHdtErr
-
-        data.close()
+        # Use threshold of |dHdt| > 1.0 to determine invalid data
+        mask = np.abs(dHdt) > 1.0
+        # Assign 100% uncertainty where data is missing
+        dHdtErr = dHdtErr.where(~mask, 1.0)
```

Comment:
Is the tilde correct here? Won't this overwrite all the good data and leave the missing/bad data in place?

Reply:
Yes, it is (although I agree it's counterintuitive). If you check the `xr.DataArray.where` docstring: 

> other : scalar, DataArray, Dataset, or callable, optional
>     Value to use for locations in this object where ``cond`` is False.
>     By default, these locations are filled with NA. If a callable, it must
>     expect this object as its only parameter. 

So we need to make the mask `False` inorder for the replacing to work, which is why the tilde is there. I got burned in a first round of testing when I left if off. 

I could switch to using `xr.where`, which has the opposite behavior (i.e. `True` values are replaced), if that's more intuitive to read.

Reply:
`xr.where` is definitely more intuitive, but given that this works and we now have documentation of this confusion, I'm fine with leaving it as-is.